crates/oxc_linter/src/rules/typescript/triple_slash_reference.rs

```diff
@@ -0,0 +1,277 @@
+struct TripleSlashReferenceDiagnostic(String, #[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct TripleSlashReference(Box<TripleSlashReferenceConfig>);
+
+#[derive(Debug, Clone, Default)]
+pub struct TripleSlashReferenceConfig {
+    lib: LibOption,
+    path: PathOption,
+    types: TypesOption,
+}
+#[derive(Debug, Default, Clone, PartialEq)]
+enum LibOption {
+    #[default]
+    Always,
+    Never,
+}
+#[derive(Debug, Default, Clone, PartialEq)]
+enum PathOption {
+    Always,
+    #[default]
+    Never,
+}
+#[derive(Debug, Default, Clone, PartialEq)]
+enum TypesOption {
+    Always,
+    Never,
+    #[default]
+    PreferImport,
+}
+
+impl std::ops::Deref for TripleSlashReference {
+    type Target = TripleSlashReferenceConfig;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallow certain triple slash directives in favor of ES6-style import declarations.
+    ///
+    /// ### Why is this bad?
+    /// Use of triple-slash reference type directives is generally discouraged in favor of ECMAScript Module imports.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// /// <reference lib="code" />
+    /// globalThis.value;
+    /// ```
+    TripleSlashReference,
+    correctness
+);
+
+impl Rule for TripleSlashReference {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let options: Option<&serde_json::Value> = value.get(0);
+        Self(Box::new(TripleSlashReferenceConfig {
+            lib: options
+                .and_then(|x| x.get("lib"))
+                .and_then(serde_json::Value::as_str)
+                .map_or_else(LibOption::default, |value| match value {
+                    "always" => LibOption::Always,
+                    "never" => LibOption::Never,
+                    _ => LibOption::default(),
+                }),
+            path: options
+                .and_then(|x| x.get("path"))
+                .and_then(serde_json::Value::as_str)
+                .map_or_else(PathOption::default, |value| match value {
+                    "always" => PathOption::Always,
+                    "never" => PathOption::Never,
+                    _ => PathOption::default(),
+                }),
+            types: options
+                .and_then(|x| x.get("types"))
+                .and_then(serde_json::Value::as_str)
+                .map_or_else(TypesOption::default, |value| match value {
+                    "always" => TypesOption::Always,
+                    "never" => TypesOption::Never,
+                    "prefer-import" => TypesOption::PreferImport,
+                    _ => TypesOption::default(),
+                }),
+        }))
+    }
+    fn run_once(&self, ctx: &LintContext) {
+        let mut import_source_set = HashSet::new();
+        for node in ctx.nodes().iter() {
```

Comment:
iterating nodes because we need to access `TSImportEqualsDeclaration`.
From my understanding, ast node is the only way to get  `TSImportEqualsDeclaration` while `ImportDeclaration` can also be accessed from `ctx.semantic().module_record().import_entries`